api/v1alpha1/preflightvalidation_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VerificationTrue    string = "True"
+	VerificationFalse   string = "False"
+	VerificationStageImage   string = "Image"
+	VerificationStageBuild string = "Build"
+)
+
+// PreflightValidationSpec describes the desired state of the resource, such as the OCP image
+// that SR CRs need to be verified against and the debug configuration of the logs
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
+type PreflightValidationSpec struct {
```

Comment:
Shall we also give this spec a list of labels like a node would have? Then it really can behave like a node and be targeted by some `Modules` only like a node would.

Reply:
Not sure i get the idea... Preflight verifies Modules and builds images , if need be, it does not deploy anything permanent onto the nodes

Reply:
Right now, the `PreflightValidation` will verify all modules with a given kernel version, including `Modules` that may not have a valid mapping.
If we added labels to that object, we could restrict that validation to some Modules only (through their `selector` field).

Reply:
as  discussed, we won't use labels in the Preflight